### PR TITLE
Add demo projects for Web3 scenarios

### PR DESCRIPTION
**Pull request by DemesneGH**
This PR introduces a new `projects/` directory dedicated to demo projects that showcase real-world applications. These projects are more complex and feature-rich than the examples in the `examples/` directory. 
Additionally, we are releasing a basic ETH wallet project for the Web3 scenario.

As outlined in `projects/README.md`, our goal is to provide developers with practical reference implementations that demonstrate how to effectively use the SDK in real-world scenarios. Currently, we are focusing on Web3-related projects, with plans to expand to additional scenarios in the future.
We believe these demo projects will be valuable resources for developers looking to build TAs tailored to real-world requirements.

### CODE REVIEW

**Comment by b49020** [comment]
Thanks @DemesneGH for this PR. It's really nice to see real world use-cases supported upstream. Care to write some tests for this project?

**Comment by ivila** [comment]
Actually during my development, I will use git path for teaclave dependencies, it's more convenient as can be use cross my team (and we use no_std).
For example, when I want to use optee-teec, I just type the following:
``` shell
cargo add optee_teec --git https://github.com/apache/incubator-teaclave-trustzone-sdk.git --branch main
```
And in cargo it will:
```toml
[package]
name = "demo"
version = "0.1.0"
edition = "2021"

[dependencies]
optee-teec = { git = "https://github.com/apache/incubator-teaclave-trustzone-sdk.git", branch = "main", version = "0.2.0" }
```

**Comment by b49020** [comment]
`Acked-by: Sumit Garg <sumit.garg@linaro.org>`

**Comment by b49020** [comment]
> Actually during my development, I will use git path for teaclave dependencies, it's more convenient as can be use cross my team (and we use no_std).

Agree that's better for development. In case of deployment it's better to use released version of OP-TEE crates here:

https://crates.io/crates/optee-teec
https://crates.io/crates/optee-utee

@DemesneGH is releasing new version of crates still on your ToDo list?

**Comment by DemesneGH** [comment]
> @DemesneGH is releasing new version of crates still on your ToDo list?

Yes it is. Let me figure out the release process on `crates.io`.

**Comment by DemesneGH** [comment]
> Yes it is. Let me figure out the release process on `crates.io`.

@mssun Could you help to add my account as the owner of these 5 crates?:

https://crates.io/search?q=optee- 

Seems the operation is:
`cargo owner --add DemesneGH` (Reference: https://doc.rust-lang.org/cargo/reference/publishing.html#cargo-owner)

**Comment by DemesneGH** [comment]
> Actually during my development, I will use git path for teaclave dependencies, it's more convenient as can be use cross my team (and we use no_std).

@ivila BTW could you share some information about what your team is working on with this SDK, such as the specific scenario? We're gathering real world use cases and would greatly appreciate your input.

**Comment by mssun** [comment]
> > Yes it is. Let me figure out the release process on `crates.io`.
> 
> @mssun Could you help to add my account as the owner of these 5 crates?:
> 
> https://crates.io/search?q=optee-
> 
> Seems the operation is: `cargo owner --add DemesneGH` (Reference: https://doc.rust-lang.org/cargo/reference/publishing.html#cargo-owner)

Done. Please check and let me know if it's okay.

**Comment by DemesneGH** [comment]
> Done. Please check and let me know if it's okay.

It works, thanks! @mssun

**Comment by ivila** [comment]
> > Actually during my development, I will use git path for teaclave dependencies, it's more convenient as can be use cross my team (and we use no_std).
> 
> @ivila BTW could you share some information about what your team is working on with this SDK, such as the specific scenario? We're gathering real world use cases and would greatly appreciate your input.

@DemesneGH Sorry I miss your response, we are using it for trusty computation, for example, PIR(private information retrieval), deep learning. We used to develop by C, and it's painful as we always met problem like memory leak or memory fault(use after free), use teaclave really helps a lot.

**Comment by DemesneGH** [comment]
> @DemesneGH Sorry I miss your response, we are using it for trusty computation, for example, PIR(private information retrieval), deep learning. We used to develop by C, and it's painful as we always met problem like memory leak or memory fault(use after free), use teaclave really helps a lot.

@ivila Thanks for your insights, issues, and PRs over the past few days. We really need polish the Rust SDK better for real-world use cases. Please feel free to discuss any problems you encounter with us. 
Additionally, we would also appreciate contributions to the real-world examples, such as a minimal demo for deep learning, etc. If you're interested please feel free to open the PR, thanks!

**Comment by DemesneGH** [comment]
> Actually during my development, I will use git path for teaclave dependencies, it's more convenient as can be use cross my team (and we use no_std). For example, when I want to use optee-teec, I just type the following:
> 
> ```shell
> cargo add optee_teec --git https://github.com/apache/incubator-teaclave-trustzone-sdk.git --branch main
> ```
> 
> And in cargo it will:
> 
> ```toml
> [package]
> name = "demo"
> version = "0.1.0"
> edition = "2021"
> 
> [dependencies]
> optee-teec = { git = "https://github.com/apache/incubator-teaclave-trustzone-sdk.git", branch = "main", version = "0.2.0" }
> ```

> > Actually during my development, I will use git path for teaclave dependencies, it's more convenient as can be use cross my team (and we use no_std).
> 
> Agree that's better for development. In case of deployment it's better to use released version of OP-TEE crates here:
> 
> https://crates.io/crates/optee-teec https://crates.io/crates/optee-utee
> 
> @DemesneGH is releasing new version of crates still on your ToDo list?

Release 0.2.0 for 6 `optee-` crates have been published in `crates.io`: https://crates.io/search?q=optee 
Now we can import the crates by `optee-teec = "0.2.0"`. For reference the code version for these `0.2.0` release is 20e698863932d395b48002e5fb126491e0caef0d .

The crates will be updated accordingly with each future release (the next planned release is `0.3.0`).

File: projects/web3/eth_wallet/README.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+to other blockchain platforms with similar requirements for secure key
+management. The implementation provides basic wallet abstractions, including:
+
+- Key Generation: Securely generating random seeds within the TEE.
+- Key Derivation: Deriving keys from seeds within the TEE.
+- Key Persistency: Storing cryptographic keys securely in the TEE.
+- Transaction Signing: Signing Ethereum transactions without exposing private
+  keys to the normal world.
+- Key Erase: Erasing keys when they are no longer needed.
+
```
[the remark this answers is not in the document]
Comment (DemesneGH) [reply projects/web3/eth_wallet/README.md:18]: Doc updated, please check, thanks!

File: projects/web3/eth_wallet/ta/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        Command::DeriveAddress => process(serialized_input, derive_address),
+        Command::SignTransaction => process(serialized_input, sign_transaction),
+        _ => bail!("Unsupported command"),
+    }
+}
+
+#[ta_invoke_command]
+fn invoke_command(cmd_id: u32, params: &mut Parameters) -> optee_utee::Result<()> {
+    dbg_println!("[+] TA invoke command");
+    let mut p0 = unsafe { params.0.as_memref().unwrap() };
```
[the remark this answers is not in the document]
Comment (DemesneGH) [reply projects/web3/eth_wallet/ta/src/main.rs]: done

File: projects/web3/eth_wallet/ta/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    let mnemonic = wallet.get_mnemonic()?;
+    dbg_println!("[+] Wallet ID: {:?}", wallet_id);
+
+    let secure_object: Vec<u8> = wallet.try_into()?;
+    save_in_secure_storage(wallet_id.as_bytes(), &secure_object)?;
+    dbg_println!("[+] Wallet saved in secure storage");
+
+    Ok(proto::CreateWalletOutput {
+        wallet_id,
+        mnemonic,
```
[the remark this answers is not in the document]
Comment (DemesneGH) [reply projects/web3/eth_wallet/ta/src/main.rs:82]: doc updated

**Pull request merged**